In [1]:
import numpy as np
import pandas as pd
import 

In [14]:
ages = [14, 12, 11, 10, 8, 6, 8]
def brady_bunch(ages):
    print('mean: ', np.mean(ages))
    print('median: ', np.median(ages))
    print('variance: ', np.var(ages))
    print('standard deviation: ', np.std(ages))
    print('standard error: ', np.std(ages)/np.sqrt(len(ages) - 1))
    
    

In [10]:
brady_bunch(ages)

mean:  9.857142857142858
median:  10.0
variance:  6.408163265306122
standard deviation:  2.531435020952764
standard error:  1.0334540197243194


mean and median are close enough to use mean as the indicator for central tendency 
for the same reason, standard deviation is the proper indicator for variance

In [12]:
new_ages = [14, 12, 11, 10, 8, 7, 8]
brady_bunch(new_ages)

mean:  10.0
median:  10.0
variance:  5.428571428571429
standard deviation:  2.32992949004287
standard error:  0.951189731211342


In [13]:
get_rid_of_oliver = [14, 12, 11, 10, 8, 7, 1]
brady_bunch(get_rid_of_oliver)

mean:  9.0
median:  10.0
variance:  15.428571428571429
standard deviation:  3.927922024247863
standard error:  1.6035674514745466


because of the outlier, we will use median instead of mean as it is a better indicator of central tendency

In [15]:
percent_fans = [20, 23, 17, 5]
percent_median = np.median(percent_fans)
print(percent_median)

18.5


because sciphi fanatic is an outlier in the dataset, then we use the median to determine the proper percent of brady bunch fans. Therefore the amount of fans is approximately 18.5%